frontend/src/modules/Tables/components/TableFilters.js

```diff
@@ -0,0 +1,215 @@
+  };
+
+  const deleteDataFilter = async (filterUri) => {
+    try {
+      const response = await client.mutate(
+        deleteTableDataFilter({
+          filterUri: filterUri
+        })
+      );
+      if (!response.errors) {
+        enqueueSnackbar('Data filter removed from table', {
+          anchorOrigin: {
+            horizontal: 'right',
+            vertical: 'top'
+          },
+          variant: 'success'
+        });
+        fetchItems();
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (e) {
+      dispatch({ type: SET_ERROR, error: e.message });
+    }
+  };
+
+  const fetchItems = useCallback(async () => {
+    try {
+      const response = await client.query(
+        listTableDataFilters({
+          tableUri: table.tableUri,
+          filter: filter
+        })
+      );
+      if (!response.errors) {
+        setItems(response.data.listTableDataFilters);
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (e) {
+      dispatch({ type: SET_ERROR, error: e.message });
+    } finally {
+      setLoading(false);
+    }
+  }, [client, dispatch, filter, table.tableUri]);
+
+  useEffect(() => {
+    if (client) {
+      fetchItems().catch((e) =>
+        dispatch({ type: SET_ERROR, error: e.message })
+      );
+    }
+  }, [client, filter.page, fetchItems, dispatch]);
+
+  if (loading) {
+    return <CircularProgress />;
+  }
+
+  return (
+    <Box>
+      <Box sx={{ mt: 3 }}>
+        <Card>
+          <CardHeader
+            action={<RefreshTableMenu refresh={fetchItems} />}
+            title={
+              <Box>
+                <SupervisedUserCircleRounded style={{ marginRight: '10px' }} />{' '}
+                Data Filters
+              </Box>
+            }
+          />
+          <Divider />
+          <Box
+            sx={{
+              alignItems: 'center',
+              display: 'flex',
+              flexWrap: 'wrap',
+              m: -1,
+              p: 2
+            }}
+          >
+            <Grid item md={10} sm={6} xs={12}>
+              <Box
+                sx={{
+                  m: 1,
+                  maxWidth: '100%',
+                  width: 500
+                }}
+              >
+                <TextField
+                  fullWidth
+                  InputProps={{
+                    startAdornment: (
+                      <InputAdornment position="start">
+                        <SearchIcon fontSize="small" />
+                      </InputAdornment>
+                    )
+                  }}
+                  onChange={handleInputChange}
+                  onKeyUp={handleInputKeyup}
+                  placeholder="Search"
+                  value={inputValue}
+                  variant="outlined"
+                />
+              </Box>
+            </Grid>
+            <Grid item md={2} sm={6} xs={12}>
+              <Button
```

Comment:
very very nit: disable this button if no rows are selected or no row expressions are saved

Reply:
I think this would be for `Add Data Filter` Button on `TableDataFilterAddForm` and not for this button here right?

Reply:
udpated TableDataFilterAddForm `Add Data Filter` to be disabled unless selected cols or row expression